components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -73,7 +75,7 @@ class IntentProcessor(
             else -> {
                 val url = extraText.split(" ").find { it.isUrl() }
                 if (url != null) {
-                    val session = createSession(url, source = Source.ACTION_SEND)
+                    val session = createSession(url, private = isPrivate, source = Source.ACTION_SEND)
                     sessionUseCases.loadUrl.invoke(url, session)
```

Comment:
Instead of adding `openNewTab` and `isPrivate` parameters we probably should just replace this with a `LoadUrlUseCase`. This allows an app to specify `AddNewTabUseCase`, `AddNewPrivateTabUseCase` or `DefaultLoadUrlUseCase`.

Reply:
Are you suggesting I make a loadUrlCase a parameter (to replace `openNewTab` and `isPrivate`) that is passed in with the constructor that is then used to open sessions with the createSession method?

Reply:
Would that work? We'd need to pass the `source` along though.